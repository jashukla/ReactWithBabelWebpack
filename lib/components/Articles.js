import React from 'react';
import Article from './Article'

const Articles = (props)=>{
    return(
        <div>
            {Object.values(props.articles).map(article =>
                <Article
                    key={article.id}
                    article = {article}
                    actions = {props.articleActions}
                />
            )}
        </div>
    )
}

export default Articles;