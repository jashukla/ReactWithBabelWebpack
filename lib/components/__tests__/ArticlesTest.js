import React from 'react';
import renderer from 'react-test-renderer';
import Articles from '../Articles';

describe('test Acrticles component',()=>{

    const testProps = {
        articles :{
            a:{id:'a'},
            b:{id:'b'}
        },
       articleActions :{
           lookupAuthor : jest.fn(()=>({})),
       }
    }

    it ('renders correctly ',()=>{
        const tree = renderer.create(
            <Articles
                {...testProps}
            ></Articles>
        ).toJSON();
        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot();
    })

})