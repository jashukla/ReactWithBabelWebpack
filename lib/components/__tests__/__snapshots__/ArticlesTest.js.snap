// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Acrticles component renders correctly  1`] = `
<div>
  <div>
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
